tt_metal/impl/debug/sanitize_noc_host.hpp

```diff
@@ -2,15 +2,38 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include <cstdint>
+
 #include "noc/noc_parameters.h"
 #include "noc/noc_overlay_parameters.h"
 
 #pragma once
 
+#undef MEM_L1_BASE
+#undef MEM_ETH_BASE
+constexpr std::uint32_t MEM_L1_BASE = 0x0;
+constexpr std::uint32_t MEM_ETH_BASE = 0x0;
+
+namespace {
+  inline std::uint32_t compute_size(const tt::tt_metal::Hal& hal, tt::tt_metal::HalProgrammableCoreType core_type) {
+    return hal.get_dev_size(core_type, tt::tt_metal::HalL1MemAddrType::UNRESERVED) +
+           hal.get_dev_addr(core_type, tt::tt_metal::HalL1MemAddrType::UNRESERVED);
+  }
+  inline std::uint32_t mem_l1_size(const tt::tt_metal::Hal& hal) {
+    return compute_size(hal, tt::tt_metal::HalProgrammableCoreType::TENSIX);
+  }
+  inline std::uint32_t mem_eth_size(const tt::tt_metal::Hal& hal) {
+    if (hal.get_arch() == tt::ARCH::GRAYSKULL) {
+      return 0;
+    }
+    return compute_size(hal, tt::tt_metal::HalProgrammableCoreType::IDLE_ETH);
```

Comment:
the compute size should be behind the hal 100%, ie, hal should return 0 for GS

Reply:
Thanks for the feedback, moved to hal now.